core/commands/stat_dht.go

```diff
@@ -67,12 +69,62 @@ This interface is not stable and may change from release to release.
 			dhts = []string{"wan", "lan"}
 		}
 
+	dhttypeloop:
 		for _, name := range dhts {
 			var dht *dht.IpfsDHT
+
+			var separateClient bool
+			if nd.DHTClient != nd.DHT {
+				separateClient = true
+			}
+
 			switch name {
 			case "wan":
+				if separateClient {
+					client, ok := nd.DHTClient.(*fullrt.FullRT)
+					if !ok {
+						return cmds.Errorf(cmds.ErrClient, "could not generate stats for the WAN DHT client type")
+					}
+					peerMap := client.Stat()
+					buckets := make([]dhtBucket, 1)
```

Comment:
nit: not needed here, just declare in-line when constructing `dhtStat` line 113
```go
dhtStat{
    Name:    name,
    Buckets: []dhtBucket{*b},
}
```

Reply:
thx, just going to leave it for now (don't want to think too hard about if this change is worth while). Either way we should really be showing the peers in multiple buckets just like the regular client, and we can do that it was just a bit of a pain so I skipped it for now.